scripts/redis/scan_redis.py

```diff
@@ -60,10 +72,27 @@ def run():
         for key in data:
             key_type = client.type(key)
             size = get_size(client, key, key_type)
+
+            # ttl() returns None in redis 2.x and -1 in redis 3.x for
+            # keys that don't have an expiration.
             ttl = client.ttl(key)
```

Comment:
Can we bulk-get ttl values with a pipeline? That's probably much more performant both for redis and the script. Same with setting new below.

Reply:
Don't want to risk a big refactor like that right now. I don't really care about the script performance and I'm not sure what is better bigger pipelines that take longer to run versus many smaller commands that allow other processes to easily interleave their requests in between this script's requests.